### PR TITLE
add placement rule for mhlo_disc::CustomCallV2Op

### PR DESCRIPTION
**Pull request by wyzero**

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/mhlo_placer.cc [hunk 1]
```diff
@@ -35,6 +35,17 @@ using placement_utils::PlacementType;
...
 namespace {
 
+PlacementType toPlacementType(StringRef s, bool default_is_gpu = false) {
+  // h -> host (CPU)
+  // d -> device (e.g. GPU)
+  // x -> xpu (same as the default device. x will be CPU is default is CPU,
+  // otherwise is GPU) s -> shape operand, usually placed on CPU as well.
+  if (s == "h" || s == "s") return PlacementType::CPU;
+  if (s == "d") return PlacementType::GPU;
+  assert(s == "x");
```
Comment (minminsun) [inline tao_compiler/mlir/disc/transforms/mhlo_placer.cc:45]: nit: assert is ignored in release build.

Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/mhlo_placer.cc:45]: yeah，we use the same strategy as LLVM.

File: tao_compiler/mlir/disc/transforms/mhlo_placer.cc [hunk 2]
```diff
@@ -35,6 +35,17 @@ using placement_utils::PlacementType;
 namespace disc_ral {
 namespace {
 
+PlacementType toPlacementType(StringRef s, bool default_is_gpu = false) {
+  // h -> host (CPU)
+  // d -> device (e.g. GPU)
+  // x -> xpu (same as the default device. x will be CPU is default is CPU,
+  // otherwise is GPU) s -> shape operand, usually placed on CPU as well.
```
Comment (qiuxiafei) [inline tao_compiler/mlir/disc/transforms/mhlo_placer.cc:42]: nnnnnit:  a new line for s?

**Review by qiuxiafei — APPROVED** [review]
LGTM

File: tao_compiler/mlir/disc/transforms/mhlo_placer.cc (hunk 2 above)
Comment (wyzero) [reply tao_compiler/mlir/disc/transforms/mhlo_placer.cc:42]: It's re-formatted by clang-format.. I'll fix this in the next PR.

**Pull request merged**